### DISCUSSION TITLE
Text Selection on components utilizing usePress underneath

### DISCUSSION THREAD

**Discussion by omacranger** [adobe/react-spectrum#3761] [Q&A]
Hey all, we've had some individuals report frustrating when attempting to select text on something that leverages usePress underneath (which appears to have allowTextselection) disabled by default. For example, something like useBreadcrumbItem  -> useLink -> usePress disables text selection by default. For clickable items, that makes since (links wouldn't really support that). But for the 'current' item, which is generally rendered as text / non-interactive, it prevents selection on the page. Are there any workarounds? I see the restoreTextSelection helpers are not exported / used internally only.

**Comment by LFDanLu** [comment]
A quick and dirty workaround would be to delete some of the event handlers returned by useBreadcrumbItem (aka the click and pointer handlers) since those are responsible for disabling text selection on click/touch. The team is discussing what to do at the hook level but there are some additional wrinkles there.

**Reply by omacranger** [reply]
I did consider working around the event listeners but it seemed flaky and I'd have to continue to keep those props updated as the library changed. I poked around and saw the source code has a flag for 'h[1-6]'  `elementTypes` which is more or less achieves what I was looking for. Perhaps there would be a way to simply flag link / link-type components vs non-link components (span, h tags, etc)?

**Reply by LFDanLu** [reply]
Perhaps we could omit the press handlers if an item was `isCurrent` in useBreadcrumbItem, but the team is still discussing if something should be done for disabled links in useLink.
